hazelcast/src/main/java/com/hazelcast/spi/InvocationBuilder.java

```diff
@@ -265,5 +268,20 @@ public InvocationBuilder setExecutionCallback(ExecutionCallback<Object> executio
         return this;
     }
 
+    /**
+     * Sets the {@code callbackExecutor} in which the {@link #executionCallback} will be run
+     *
+     * @param callbackExecutor the callbackExecutor in which the callback will be run
+     * @return the updated InvocationBuilder.
+     * @throws java.lang.IllegalStateException if an callbackExecutor has already been set.
+     */
+    public InvocationBuilder setCallbackExecutor(Executor callbackExecutor) {
+        if (this.callbackExecutor != null) {
```

Comment:
Should there not be a check if the calllbackExecutor is null? I don't think that is a desirable situation. Or otherwise it needs to be documented.